.claude/scripts/format.sh

```diff
@@ -6,13 +6,102 @@
 
 set -euo pipefail
 
+# A variable to memoize the command for canonicalizing paths.
+_CANONICALIZE_CMD=""
+
+# canonicalize_path resolves a path to its absolute, canonical form.
+# It tries 'realpath', 'readlink -f', 'python3', 'python', or 'perl' in order.
+# The chosen command is memoized to avoid repeated checks.
+# If none of these are available, it returns an empty string.
+canonicalize_path() {
+	local path_to_resolve="$1"
+
+	# If we haven't determined a command yet, find one.
+	if [[ -z "$_CANONICALIZE_CMD" ]]; then
+		if command -v realpath >/dev/null 2>&1; then
+			_CANONICALIZE_CMD="realpath"
+		elif command -v readlink >/dev/null 2>&1 && readlink -f . >/dev/null 2>&1; then
+			_CANONICALIZE_CMD="readlink"
+		elif command -v python3 >/dev/null 2>&1; then
+			_CANONICALIZE_CMD="python3"
+		elif command -v python >/dev/null 2>&1; then
+			_CANONICALIZE_CMD="python"
+		elif command -v perl >/dev/null 2>&1; then
+			_CANONICALIZE_CMD="perl"
```

Comment:
Why do we need to run python or perl interpreters at all here?

Reply:
Those are fallbacks in case realpath or readlink are not available.

While working with CC hooks, I noticed that their file paths are wonky. Sometimes, they are relative, and sometimes, they are absolute, often a misformulation of both.

So this is just in case someone develops coder on an env that doesn’t have realpath or readlink.

Reply:
> So this is just in case someone develops coder on an env that doesn’t have realpath or readlink.

Can you give an example? I see these available in Nix and they're also available on darwin via Brew / `coreutils`.

Reply:
windows

Reply:
Unless we don't care about Windows development, then I'm happy to zap this